portmap/core/views.py

```diff
@@ -68,6 +73,12 @@ def create_datatype(datatype_object):
                 'datatypes': datatypes
             }
 
+def csrf_token(request):
```

Comment:
Does this introduce a security flaw?  because it's intended to prevent cross-site request forgery, could it be used by an attacker to grab a valid CSRF from us and use it in the cross-site URL sent to the browser?

* Since CSRF tokens are unique to a session and checked against the session, I think the attacker cannot get a valid CSRF for anybody but their own session.  they can get that anyway. 

* Even if this is a flaw, there's very little at risk in our threat model - this is a public page so an attacker could get a valid CSRF token from the public page and use that. 

* maybe the right thing to do anyway since this is a public view is use [csrf_exempt()](https://docs.djangoproject.com/en/5.0/ref/csrf/#django.views.decorators.csrf.csrf_exempt).  We have views in the backend admin that we'd like to protect with CSRF - although chances are very low that somebody would target a staff member for django admin access, might as well protect as if there are going to be attackers.  But a public form that doesn't even implement session tracking, seems like it could be an exception to the CSRF broad brush.

Reply:
I meant to note this in my PR, but for the reasons you mentioned, I don't believe this is any less secure than the existing views that include tokens in the response HTML. Just like with those requests, browsers won't allow a non-opaque, cross-origin request to this route, so a malicious site in a user's browser still won't be able to inspect responses from this route (and again as you mentioned, if this was possible, they could already get a token just by loading any of the views and extracting it from the HTML).

I agree that it would probably be okay to exempt this flow since it just protects leaving anonymous feedback, and I'm having a hard time imagining how this could be vulnerable to *cross-site* shenanigans. But I don't think it hurts to leave it on either.